src/server/router/vehicle.router.ts

```diff
@@ -16,4 +18,46 @@ export const vehicleRouter = createRouter().mutation(
       });
     },
   }
-);
+)
+.query(
+  'request-vehicle-make',
+  {
+    async resolve() {
+      const response = await fetch(
+        carbonVehicleMakeRequest()
+      );
+      let vehicleOptions: VehicleMakeSearch[] = []
+
+      return response.json().then((data) => {
+        const vehicles = data as VehicleMakeDataResponse[];
+        vehicleOptions = vehicles.map((vehicle: VehicleMakeDataResponse) => {
+          return {name: vehicle.data.attributes.name, value: vehicle.data.id}
+        })
+        return vehicleOptions;
+      }
+      )
+    }
+  }
+)
+.query(
+  'request-vehicle-model',
+  {
+    input: z.string(),
+    async resolve({input}) {
+      const response = await fetch(
+        // Query keeps firing on component load so added a fallback value to keep from getting invalid input errors
+        carbonVehicleModelsRequest(input ? input : '4c1e16e1-7967-4394-b3cb-15f4577dffa1')
+      );
+      let vehicleOptions: VehicleModelSearch[] = []
+
+      return response.json().then((data) => {
+        const vehicles = data as VehicleModelDataResponse[];
+        vehicleOptions = vehicles.map((vehicle: VehicleModelDataResponse) => {
+          return {name: vehicle.data.attributes.name, value: vehicle.data.id}
```

Comment:
As a suggestion here, in your tRPC query for the models, why not just return `{name: vehicle.data.attributes.year + ' ' + vehicle.data.attributes.name, ... }` in your map function? Or if that don't like that, add a line before your return in the map function `const vehicleYearAndName = [vehicle.data.attributes.year, vehicle.data.attributes.name].join(' ')` and just return `vehicleYearAndName `in the value field for `name`

Reply:
Good point added the year to the models listing! I'm going to make a separate ticket for the models data filtering so we can get this merged in sooner